.specify/memory/constitution.md

```diff
@@ -0,0 +1,139 @@
+<!--
+Sync Impact Report:
+Version Change: N/A → 1.0.0
+Initial Constitution Creation
+Added Sections:
+  - Core Principles (6 principles)
+  - Data Management Standards
+  - Development Workflow
+  - Governance
+Templates Requiring Updates:
+  ✅ .specify/templates/plan-template.md - to be validated
+  ✅ .specify/templates/spec-template.md - to be validated
+  ✅ .specify/templates/tasks-template.md - to be validated
+Follow-up TODOs: None
+-->
+
+# Agentic Neurodata Conversion Constitution
+
+## Core Principles
+
+### I. MCP-Centric Architecture
+
+All functionality MUST be exposed through standardized MCP (Model Context Protocol) tools. Direct agent invocation from external clients is prohibited. The MCP server acts as the single orchestration layer, ensuring consistent interfaces and enabling horizontal scaling.
+
+**Rationale**: Centralizing all operations through MCP tools creates a unified API surface, simplifies client integration, and allows independent scaling of internal agents.
+
+### II. Agent Specialization
+
+Each internal agent (Conversation, Conversion, Evaluation, Knowledge Graph) MUST handle a single domain of responsibility. Cross-domain operations require coordination through the MCP server, not direct agent-to-agent communication.
+
+**Rationale**: Clear separation of concerns ensures maintainability, testability, and allows independent evolution of agent capabilities.
+
+### III. Test-Driven Development (NON-NEGOTIABLE)
+
+All features MUST follow strict TDD workflow:
+1. Write tests that define expected behavior
+2. Obtain user approval on test specifications
+3. Verify tests fail appropriately
+4. Implement feature to pass tests
+5. Refactor while maintaining green tests
+
+**Rationale**: Neuroscience data conversion is mission-critical. TDD ensures correctness, prevents regressions, and provides living documentation of system behavior.
+
+### IV. Data Integrity & Validation
+
+All data conversions MUST:
+- Validate input data structure before conversion
+- Verify NWB schema compliance using NWB Inspector
```

Comment:
inspector has a validator built-in but that's not the main thing we are using it for.